### ISSUE TITLE
Hard-coded document fields in Screen::EPrint::Details

### ISSUE THREAD

**Issue by jesusbagpuss** [eprints/eprints#301]
https://github.com/eprints/eprints/blob/3.3/perl_lib/EPrints/Plugin/Screen/EPrint/Details.pm#L197-L221
Rendered fields are hard-coded.

Should use a citation format of some description..?

**Comment by patrickmcsweeney** [comment]
Yes i think it should be. In fact i think there are a lot of the places in the core code where we should be using phrases rather than building DOM in the code to give us some controller/view seperation. This should maybe be a wider discussion point

**Comment by sebastfr** [comment]
(Looking at this from far away) The three concepts (templates, citations, phrases) are overlapping, they do the same thing really so why not having a single concept. And as Patrick suggests, no html building should be in the perl code. See how other frameworks implement such concepts (drupal, django, ...) for inspiration. You'll see what they do is much more simple than eprints'. And if one day you need to move away from XML templates, it'll be easier to migrate to e.g. mustache/twig/whatever. Anyways, easy for me to say, good luck on your contributions day.

**Comment by jesusbagpuss** [comment]
@sebastfr good to see you're still keeping an eye on us :o)
The bigger MVC issue (that you were progressing with v4) is still on the masterplan (somewhere)!

**Comment by MrkGrgsn** [comment]
A while back when I was still trying to keep up with EPrints things, I whipped up a prototype of EPrints using the XSlate template system based on 3.3 https://github.com/MrkGrgsn/eprints/tree/xslate-templates.  It takes the approach of minimal changes required to introduce a template system for citations, DataObj's and templates and can be run alongside the existing UI code generation method. You're welcome to have a look and I'll write up a pull request if you want.

**Comment by denics** [comment]
I do not know if I'm late, but I would vote for moving ALL hard-coded rendering to citations. This will be very much appreciated by themers!

**Comment by patrickmcsweeney** [comment]
Hi Seb :-)

Seb is right Django (web framework in python) is an excellent example of
this done well. If you want an example closer to home then Dancer (web
framework in perl) and TemplateToolkit (templating language in perl) are
good places to get ideas. I think we really should look at addressing the
issue of the citation, phrase, template overlap. A short term (next 12
months) solution would be to simply consolidate templates, phrases and
citations into one thing still using EPC for backwards compatibility and
then slowly moving calls to the $xml and $xhtml functions out to rendering.
However this is a moderately sized job and at the moment we arent even in a
position to take accept pull requests and we really need to get there so we
can make better use of the community we have.


On Tue, Nov 24, 2015 at 11:30 AM, Denix <notifications@github.com> wrote:

> I do not know if I'm late, but I would vote for moving ALL hard-coded
> rendering to citations. This will be very much appreciated by themers!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eprints/eprints/issues/301#issuecomment-159237086>.
>



-- 

'But your intentions are beside the point, It's the outcome of your actions
that count...'

**Comment by denics** [comment]
Any more input here about moving to a framework? Catalyst, Dancer or whatever else would really improve EPrints stability I think.
